scala3doc/src/dotty/tools/dottydoc/Main.scala

```diff
@@ -32,49 +32,18 @@ object Main extends Driver {
     * how they're split).
     */
   override def process(args: Array[String], rootCtx: Context): Reporter = {
-    // split args into ours and Dotty's
-    val (dokkaStrArgs, compilerArgs) = {
-      args.partitionMap { arg =>
-        // our options start with this magic prefix, inserted by the SBT plugin
-        val magicPrefix = "--+DOC+"
-        if arg startsWith magicPrefix then
-          Left(arg stripPrefix magicPrefix)
-        else
-          Right(arg)
-      }
-    }
 
-    val (filesToCompile, ctx) = setup(compilerArgs, rootCtx)
-    given Context = ctx
 
-    // parse Dokka args
-    // note: all required args should be set with SBT settings,
-    // to make it easier to set and override them
-    val dokkaArgs = {
-      val dokkaRawArgs = new RawArgs
-      val requiredArgs = Seq(
-        "--tastyRoots", "", // hack, value is not used in SBT but required in CLI
-        // we extract some settings from Dotty options since that's how SBT passes them
-        "--name", ctx.settings.projectName.value,
-        "--projectTitle", ctx.settings.projectName.value,
-        "--dest", ctx.settings.outputDir.value.toString,
-      )
+    val (filesToCompile, ctx) = setup(args, rootCtx)
```

Comment:
@romanowski this doesn't work, we need to parse the arguments ourselves first, then "forward" them to compiler settings. Otherwise the compiler will emit errors on Scala3doc-specific args. I can take a look at that later today.